cupyx/distributed/_store.py

```diff
@@ -45,9 +54,6 @@ def __init__(self, world_size):
         self._lock = threading.Lock()
         self._current_barrier = None
 
-    def __del__(self):
-        self.stop()
```

Comment:
How about keeping `__del__`, and call `stop()` only when `_exit_mode` is not set? I think it's better to destroy the process once the communicator goes out of scope.

Reply:
done :)